scalalib/src/mill/scalalib/JavaModule.scala

```diff
@@ -41,7 +41,13 @@ trait JavaModule
     override def zincWorker: ZincWorkerModule = outer.zincWorker
     override def skipIdea: Boolean = outer.skipIdea
     override def runUseArgsFile: Target[Boolean] = T { outer.runUseArgsFile() }
+    override def sources = T.sources {
+      for (src <- outer.sources()) yield {
+        PathRef(this.millSourcePath / src.path.relativeTo(outer.millSourcePath))
```

Comment:
What if the user added additional source folders, maybe pointing outside of `outer.millSourcePath`?

Reply:
Aeems reasonable to mirror them inside `test` I think? That's what I'm finding as I port builds over, e.g. maybe someone added `src-2.13+/` or `src-js-jvm/`, and it would make perfect sense to mirror them inside the `test` submodule

We could filter these folders somehow if we really want, but IMO it's not necessary. In most common cases the only downside would be the `test` module gets a few unused-but-reasonable source folders, and worst comes to worst if someone wants to change this mirroring they can always override it.

Reply:
One more consideration is that the current approach of shadowing the  `Tests` trait does not compose. While it works for one use case in `CrossScalaModule`, you cannot stack a second use case in `PlatformScalaModule` and have them work together to define the final sources list

We also already delegate a lot of targets from test module to host module, so delegating another one would fit right in